### DISCUSSION TITLE
How to flexibly set  each variant type when creating a ua server（add_variable）

### DISCUSSION THREAD

**Discussion by ryangu18** [FreeOpcUa/opcua-asyncio#835] [Q&A] [answered]
Dear friends, I am a newbie. I am practicing using this library to create a ua server

I have a csv files, the third column is variant type, I want each tag to have different variant type, but I don't know the grammar,
![image](https://user-images.githubusercontent.com/99937589/156987552-357b1b2c-3a7a-4703-8523-fdca454a374e.png)

	for eachTag in TagList:
		
		Temp = await Param.add_variable(addspace, eachTag[0], 0.0,ua.VariantType.Double)  # 
		
		await Temp.set_writable()

I have get the value from the third column and save to eachTag[2], which is a string type. I want to change  variant type from ua.VariantType.Double to eachTag[2]. How should I modify the program?

Thanks in advance for your help.

**Comment by schroeder-** [comment, marked as answer]
Just provide a dict such as:
```

datatype_to_variant = {
   'Float':  (0.0, ua.VariantType.Double),
   'Int32': (ua.Int32(0), ua.ua.VariantType.Int32),
   ....
}

def_val, datatype = datatype_to_variant[eachTag[2]]
Temp = await Param.add_variable(addspace, eachTag[0], def_val, datatype)  # 
```

**Reply by ryangu18** [reply]
Thanks for your quick reply, this is exactly the answer I was looking for, thank you very much！
